plugins/auth-backend/src/extensions.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createExtensionPoint } from '@backstage/backend-plugin-api';
+import { ProviderFactories } from '@backstage/plugin-auth-backend';
+
+/**
+ * @alpha
+ */
+export interface AuthProviderExtensionPoint {
+  addProviderFactories(factories: ProviderFactories): void;
```

Comment:
This contract should probably accept one at a time, so `addAuthProvider(id: string, factory: AuthProviderFactory): void` or so.

Reply:
why ? the meaning of this was to add all the factories provided in the `ProviderFactories` argument, which means merging the factories with those already existing.

See the implementation: https://github.com/backstage/backstage/pull/18122/files#diff-f4495cdf6894a017f4c907754086c8ead68c23a035a47dec4f222652727e9011R43